pkg/ipcache/ipcache.go

```diff
@@ -285,6 +307,7 @@ func (ipc *IPCache) upsertLocked(
 	k8sMeta *K8sMetadata,
 	newIdentity Identity,
 	force bool,
+	fromLegacyAPI bool,
```

Comment:
minor stylistic question: does it make sense for `Upsert()` to set `newIdentity.modifiedByLegacyAPI` and drop this parameter?